client/ayon_substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -156,6 +172,110 @@ def create_image_instance(self, instance, template, outputs,
         image_instance.data["textureSetName"] = texture_set_name
         image_instance.data["textureStackName"] = stack_name
 
+        # Store color space with the instance
+        # Note: The extractor will assign it to the representation
+        colorspace = outputs[0].get("colorSpace")
+        if colorspace:
+            self.log.debug(f"{image_product_name} colorspace: {colorspace}")
+            image_instance.data["colorspace"] = colorspace
+
+        # Store the instance in the original instance as a member
+        instance.append(image_instance)
+
+    def create_image_instance_by_map_filtering(self, instance, outputs,
+                                               task_entity, map_identifier):
+        """Create a new instance per image based on map filtering.
+
+        The new instances will be of product type `image`.
+        **Only used for exporting multiple texture sets as one texture output
+
+        """
+
+        context = instance.context
+        first_filepath = outputs[0]["filepath"]
+        fnames = [os.path.basename(output["filepath"]) for output in outputs]
+        ext = os.path.splitext(first_filepath)[1]
+        assert ext.lstrip("."), f"No extension: {ext}"
+        # Function to remove textureSet from filepath
+        def remove_texture_set_token(filepath, texture_set):
+            return filepath.replace(texture_set, '')
```

Comment:
Why do we need this? Do we care about the texture set name at all?

Reply:
We need to have new filename for the single output :).